### PR TITLE
MISRA violation fix 17.7, 1.2, 8.9

### PR DESCRIPTION
**Pull request by xuelix**
Fixing MISRA violation 17.7, 1.2 and 8.9
Description
-----------

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_DNS_Cache.c [hunk 1]
```diff
@@ -173,9 +173,9 @@
         { /* Element found */
             if( xLookUp == pdTRUE )
             {
-                prvGetCacheIPEntry( uxIndex,
-                                    pulIP,
-                                    ulCurrentTimeSeconds );
+                ( void ) prvGetCacheIPEntry( uxIndex,
```
Comment (AniruddhaKanhere) [inline source/FreeRTOS_DNS_Cache.c:176]: Maybe we should add a comment about why the result is being discarded here and why is it safe to do so.
Something along the lines of:
```
/* This statement can only be reached when xResult is true; which
 * implies that the entry is present and a 'get' operation will result
 * in success. Therefore, it is safe to ignore the return value of the
 * below function. */
```

File: source/FreeRTOS_IP_Utils.c [hunk 2]
```diff
@@ -321,6 +321,10 @@ BaseType_t xIsCallingFromIPTask( void )
 /**
  * @brief Process a 'Network down' event and complete required processing.
  */
+/* MISRA Ref 8.9.1 [File scoped variables] */
+/* More details at: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/MISRA.md#rule-89 */
+/* coverity[misra_c_2012_rule_8_9_violation] */
+/* coverity[single_use] */
```
Comment (AniruddhaKanhere) [inline source/FreeRTOS_IP_Utils.c:327]: Does this suppress the violation in the report? If I remember correctly, we had found that doing this doesn't help.

**Review by AniruddhaKanhere — COMMENTED** [review]
Thanks for this PR @xuelix.
I have a couple of comments. Can you please take a look?

File: source/FreeRTOS_IP_Utils.c (hunk 2 above)
Comment (xuelix) [reply source/FreeRTOS_IP_Utils.c:327]: It does suppress the violation in the report

File: source/FreeRTOS_DNS_Cache.c (hunk 1 above)
Comment (xuelix) [reply source/FreeRTOS_DNS_Cache.c:176]: We have too many (void) in front of functions (e.g. thousands of them including memcpy) .  I don't think we need to add for one specific function calls.

**Review by n9wxu — APPROVED** [review]

**Pull request merged**